rfcs/convergence/contract-for-trigger-pattern.md

```diff
@@ -0,0 +1,134 @@
+# RFC: Creating a structured contract for Trigger - like patterns
+
+@ling1726
+
+## Problem statement
+
+```tsx
+<Tooltip>
+  <Button />
+</Tooltip>
+
+<Menu>
+  <MenuTrigger>
+    <Button />
+  </MenuTrigger>
+</Menu>
+
+<Popover>
+  <PopoverTrigger>
+    <Button />
+  </PopoverTrigger>
+</Popover>
```

Comment:
I am not sure that we discussed this before in any RFC, but I would like to callout that we also should ensure that composition works properly. I mean a scenario below when `Button` is a trigger for `Menu` and has a `Tooltip` for example:

```tsx
<Menu>
  <MenuTrigger>
    <Tooltip>
      <Button />
    </Tooltip>
  </MenuTrigger>
</Menu>
```